Multiple tests are failing on Fedora

<!--
Thanks for wanting to report an issue you've found in uvloop. Please fill in
the template below.

It will be much easier for us to fix the issue if a test case that reproduces
the problem is provided, with clear instructions on how to run it.

Thank you!
-->

* **uvloop version**: 0.12.2
* **Python version**: 3.7.3
* **Platform**: Fedora 31 x86_64
* **Can you reproduce the bug with `PYTHONASYNCIODEBUG` in env?**: yes

```python
BUILDSTDERR: context: {'message': 'Future exception was never retrieved', 'exception': Error([('SSL routines', 'SSL_renegotiate', 'wrong ssl version')]), 'future': <Future finished exception=Error([('SSL routines', 'SSL_renegotiate', 'wrong ssl version')])>}
BUILDSTDERR: Traceback (most recent call last):
BUILDSTDERR:   File "uvloop/loop.pyx", line 2276, in uvloop.loop.Loop.call_exception_handler
BUILDSTDERR:     self._exception_handler(self, context)
BUILDSTDERR:   File "/builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py", line 78, in loop_exception_handler
BUILDSTDERR:     self.loop.default_exception_handler(context)
BUILDSTDERR: AttributeError: 'NoneType' object has no attribute 'default_exception_handler'
BUILDSTDERR: ok
BUILDSTDERR: test_remote_shutdown_receives_trailing_data (test_tcp.Test_UV_TCPSSL) ... ok
BUILDSTDERR: test_renegotiation (test_tcp.Test_UV_TCPSSL) ... Exception in thread test-server:
BUILDSTDERR: Traceback (most recent call last):
BUILDSTDERR:   File "/builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py", line 477, in _run
BUILDSTDERR:     self._handle_client(conn)
BUILDSTDERR:   File "/builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py", line 486, in _handle_client
BUILDSTDERR:     self._prog(TestSocketWrapper(sock))
BUILDSTDERR:   File "/builddir/build/BUILD/uvloop-0.12.2/tests/test_tcp.py", line 2150, in server
BUILDSTDERR:     conn.renegotiate()
BUILDSTDERR:   File "/usr/lib/python3.7/site-packages/OpenSSL/SSL.py", line 1902, in renegotiate
BUILDSTDERR:     _openssl_assert(_lib.SSL_renegotiate(self._ssl) == 1)
BUILDSTDERR:   File "/usr/lib/python3.7/site-packages/OpenSSL/_util.py", line 67, in openssl_assert
BUILDSTDERR:     exception_from_error_queue(error)
BUILDSTDERR:   File "/usr/lib/python3.7/site-packages/OpenSSL/_util.py", line 54, in exception_from_error_queue
BUILDSTDERR:     raise exception_type(errors)
BUILDSTDERR: OpenSSL.SSL.Error: [('SSL routines', 'SSL_renegotiate', 'wrong ssl version')]
BUILDSTDERR: During handling of the above exception, another exception occurred:
BUILDSTDERR: Traceback (most recent call last):
BUILDSTDERR:   File "/usr/lib64/python3.7/threading.py", line 917, in _bootstrap_inner
BUILDSTDERR:     self.run()
BUILDSTDERR:   File "/builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py", line 446, in run
BUILDSTDERR:     self._run()
BUILDSTDERR:   File "/builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py", line 483, in _run
BUILDSTDERR:     self._test._abort_socket_test(ex)
BUILDSTDERR:   File "/builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py", line 227, in _abort_socket_test
BUILDSTDERR:     self.fail(ex)
BUILDSTDERR:   File "/usr/lib64/python3.7/unittest/case.py", line 680, in fail
BUILDSTDERR:     raise self.failureException(msg)
BUILDSTDERR: AssertionError: [('SSL routines', 'SSL_renegotiate', 'wrong ssl version')]
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a94e208>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a94e698>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a94ec88>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a94f298>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a7699f8>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a769ec8>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a76a4f8>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a76ab28>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a76b158>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a76b788>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a8542b8>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a854dc8>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a8558d8>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a884968>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a885478>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a885f88>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a90c468>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a90ce78>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a90d988>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a803bc8>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a8046b8>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a8051c8>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a90e2f8>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:97: ResourceWarning: unclosed resource <TCPTransport closed=False reading=False 0x55a33a90ee08>
BUILDSTDERR:   self.loop.close()
BUILDSTDERR: ERROR
BUILDSTDERR: test_shutdown_cleanly (test_tcp.Test_UV_TCPSSL) ... /builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py:112: ResourceWarning: unclosed transport <uvloop.loop._SSLProtocolTransport object>
BUILDSTDERR:   gc.collect()
BUILDSTDERR: ok
BUILDSTDERR: test_shutdown_timeout (test_tcp.Test_UV_TCPSSL) ... ok
BUILDSTDERR: test_shutdown_timeout_handler_leak (test_tcp.Test_UV_TCPSSL) ... ok
BUILDSTDERR: test_ssl_connect_accepted_socket (test_tcp.Test_UV_TCPSSL) ... ok
BUILDSTDERR: test_ssl_handshake_timeout (test_tcp.Test_UV_TCPSSL) ... ok
BUILDSTDERR: test_start_tls_client_buf_proto_1 (test_tcp.Test_UV_TCPSSL) ... ok
BUILDSTDERR: test_start_tls_client_corrupted_ssl (test_tcp.Test_UV_TCPSSL) ... ok
BUILDSTDERR: test_start_tls_client_reg_proto_1 (test_tcp.Test_UV_TCPSSL) ... ok
BUILDSTDERR: test_start_tls_server_1 (test_tcp.Test_UV_TCPSSL) ... ok
BUILDSTDERR: test_start_tls_slow_client_cancel (test_tcp.Test_UV_TCPSSL) ... ok
BUILDSTDERR: test_start_tls_wrong_args (test_tcp.Test_UV_TCPSSL) ... ok
BUILDSTDERR: test_write_to_closed_transport (test_tcp.Test_UV_TCPSSL) ... ok
BUILDSTDERR: test_duplicate_methods (test_testbase.TestBaseTest) ... ok
BUILDSTDERR: test_duplicate_methods_parent_1 (test_testbase.TestBaseTest) ... ok
BUILDSTDERR: test_duplicate_methods_parent_2 (test_testbase.TestBaseTest) ... ok
BUILDSTDERR: test_create_datagram_endpoint_addrs_ipv4 (test_udp.Test_AIO_UDP) ... ok
BUILDSTDERR: test_create_datagram_endpoint_addrs_ipv6 (test_udp.Test_AIO_UDP) ... ok
BUILDSTDERR: test_create_datagram_endpoint_ipv6_family (test_udp.Test_AIO_UDP) ... ok
BUILDSTDERR: test_create_datagram_endpoint_sock (test_udp.Test_AIO_UDP) ... ok
BUILDSTDERR: test_create_datagram_endpoint_sock_unix_domain (test_udp.Test_AIO_UDP) ... ok
BUILDSTDERR: test_create_datagram_endpoint_addrs_ipv4 (test_udp.Test_UV_UDP) ... ok
BUILDSTDERR: test_create_datagram_endpoint_addrs_ipv6 (test_udp.Test_UV_UDP) ... ok
BUILDSTDERR: test_create_datagram_endpoint_ipv6_family (test_udp.Test_UV_UDP) ... ok
BUILDSTDERR: test_create_datagram_endpoint_sock (test_udp.Test_UV_UDP) ... ok
BUILDSTDERR: test_create_datagram_endpoint_sock_unix_domain (test_udp.Test_UV_UDP) ... ok
BUILDSTDERR: test_create_datagram_endpoint_wrong_sock (test_udp.Test_UV_UDP) ... ok
BUILDSTDERR: test_send_after_close (test_udp.Test_UV_UDP) ... ok
BUILDSTDERR: test_udp_sendto_dns (test_udp.Test_UV_UDP) ... ok
BUILDSTDERR: test_create_unix_connection_2 (test_unix.Test_AIO_Unix) ... ok
BUILDSTDERR: test_create_unix_connection_3 (test_unix.Test_AIO_Unix) ... ok
BUILDSTDERR: test_create_unix_connection_4 (test_unix.Test_AIO_Unix) ... ok
BUILDSTDERR: test_create_unix_connection_5 (test_unix.Test_AIO_Unix) ... ok
BUILDSTDERR: test_create_unix_connection_6 (test_unix.Test_AIO_Unix) ... ok
BUILDSTDERR: test_create_unix_connection_open_con_sock (test_unix.Test_AIO_Unix) ... ok
BUILDSTDERR: test_create_unix_connection_open_unix_con_addr (test_unix.Test_AIO_Unix) ... ok
BUILDSTDERR: test_create_unix_connection_open_unix_con_sock (test_unix.Test_AIO_Unix) ... ok
BUILDSTDERR: test_create_unix_server_1 (test_unix.Test_AIO_Unix) ... ok
BUILDSTDERR: test_create_unix_server_2 (test_unix.Test_AIO_Unix) ... ok
BUILDSTDERR: test_create_unix_server_3 (test_unix.Test_AIO_Unix) ... ok
BUILDSTDERR: test_create_unix_server_existing_path_sock (test_unix.Test_AIO_Unix) ... ok
BUILDSTDERR: test_transport_unclosed_warning (test_unix.Test_AIO_Unix) ... skipped 'Python version must be < 3.7'
BUILDSTDERR: test_create_unix_connection_ssl_1 (test_unix.Test_AIO_UnixSSL) ... ok
BUILDSTDERR: test_create_unix_server_ssl_1 (test_unix.Test_AIO_UnixSSL) ... ok
BUILDSTDERR: test_create_unix_connection_2 (test_unix.Test_UV_Unix) ... ok
BUILDSTDERR: test_create_unix_connection_3 (test_unix.Test_UV_Unix) ... ok
BUILDSTDERR: test_create_unix_connection_4 (test_unix.Test_UV_Unix) ... ok
BUILDSTDERR: test_create_unix_connection_5 (test_unix.Test_UV_Unix) ... ok
BUILDSTDERR: test_create_unix_connection_6 (test_unix.Test_UV_Unix) ... ok
BUILDSTDERR: test_create_unix_connection_open_con_sock (test_unix.Test_UV_Unix) ... ok
BUILDSTDERR: test_create_unix_connection_open_unix_con_addr (test_unix.Test_UV_Unix) ... ok
BUILDSTDERR: test_create_unix_connection_open_unix_con_sock (test_unix.Test_UV_Unix) ... ok
BUILDSTDERR: test_create_unix_connection_pathlib (test_unix.Test_UV_Unix) ... ok
BUILDSTDERR: test_create_unix_server_1 (test_unix.Test_UV_Unix) ... ok
BUILDSTDERR: test_create_unix_server_2 (test_unix.Test_UV_Unix) ... ok
BUILDSTDERR: test_create_unix_server_3 (test_unix.Test_UV_Unix) ... ok
BUILDSTDERR: test_create_unix_server_existing_path_sock (test_unix.Test_UV_Unix) ... ok
BUILDSTDERR: test_create_unix_server_path_dgram (test_unix.Test_UV_Unix) ... ok
BUILDSTDERR: test_create_unix_server_path_stream_bittype (test_unix.Test_UV_Unix) ... ok
BUILDSTDERR: test_create_unix_server_pathlib (test_unix.Test_UV_Unix) ... ok
BUILDSTDERR: test_epollhup (test_unix.Test_UV_Unix) ... ok
BUILDSTDERR: test_transport_fromsock_get_extra_info (test_unix.Test_UV_Unix) ... ok
BUILDSTDERR: test_transport_unclosed_warning (test_unix.Test_UV_Unix) ... skipped 'Python version must be < 3.7'
BUILDSTDERR: test_create_unix_connection_ssl_1 (test_unix.Test_UV_UnixSSL) ... ok
BUILDSTDERR: test_create_unix_server_ssl_1 (test_unix.Test_UV_UnixSSL) ... ok
BUILDSTDERR: ======================================================================
BUILDSTDERR: ERROR: test_write_pipe (test_pipes.Test_UV_Pipes)
BUILDSTDERR: ----------------------------------------------------------------------
BUILDSTDERR: Traceback (most recent call last):
BUILDSTDERR:   File "/builddir/build/BUILD/uvloop-0.12.2/tests/test_pipes.py", line 144, in test_write_pipe
BUILDSTDERR:     transport, p = self.loop.run_until_complete(connect)
BUILDSTDERR:   File "uvloop/loop.pyx", line 1451, in uvloop.loop.Loop.run_until_complete
BUILDSTDERR:     return future.result()
BUILDSTDERR:   File "uvloop/loop.pyx", line 2703, in connect_write_pipe
BUILDSTDERR:     raise
BUILDSTDERR:   File "uvloop/loop.pyx", line 2700, in uvloop.loop.Loop.connect_write_pipe
BUILDSTDERR:     await waiter
BUILDSTDERR:   File "uvloop/cbhandles.pyx", line 71, in uvloop.loop.Handle._run
BUILDSTDERR:     (<method_t>self.callback)(self.arg1)
BUILDSTDERR:   File "uvloop/handles/basetransport.pyx", line 139, in uvloop.loop.UVBaseTransport._call_connection_made
BUILDSTDERR:     self._start_reading()
BUILDSTDERR:   File "uvloop/handles/stream.pyx", line 305, in uvloop.loop.UVStream._start_reading
BUILDSTDERR:     self._fatal_error(exc, True)
BUILDSTDERR:   File "uvloop/handles/basetransport.pyx", line 46, in uvloop.loop.UVBaseTransport._fatal_error
BUILDSTDERR:     raise exc
BUILDSTDERR: OSError: [Errno 107] Transport endpoint is not connected
BUILDSTDERR: ======================================================================
BUILDSTDERR: ERROR: test_flush_before_shutdown (test_tcp.Test_UV_TCPSSL)
BUILDSTDERR: ----------------------------------------------------------------------
BUILDSTDERR: Traceback (most recent call last):
BUILDSTDERR:   File "/builddir/build/BUILD/uvloop-0.12.2/tests/test_tcp.py", line 2568, in test_flush_before_shutdown
BUILDSTDERR:     self.loop.run_until_complete(client(srv.addr))
BUILDSTDERR:   File "uvloop/loop.pyx", line 1451, in uvloop.loop.Loop.run_until_complete
BUILDSTDERR:     return future.result()
BUILDSTDERR:   File "/builddir/build/BUILD/uvloop-0.12.2/tests/test_tcp.py", line 2552, in client
BUILDSTDERR:     data = await reader.readexactly(4)
BUILDSTDERR:   File "/usr/lib64/python3.7/asyncio/streams.py", line 677, in readexactly
BUILDSTDERR:     raise IncompleteReadError(incomplete, n)
BUILDSTDERR: asyncio.streams.IncompleteReadError: 0 bytes read on a total of 4 expected bytes
BUILDSTDERR: ======================================================================
BUILDSTDERR: ERROR: test_renegotiation (test_tcp.Test_UV_TCPSSL)
BUILDSTDERR: ----------------------------------------------------------------------
BUILDSTDERR: Traceback (most recent call last):
BUILDSTDERR:   File "/builddir/build/BUILD/uvloop-0.12.2/tests/test_tcp.py", line 2234, in test_renegotiation
BUILDSTDERR:     run(client)
BUILDSTDERR:   File "/builddir/build/BUILD/uvloop-0.12.2/tests/test_tcp.py", line 2229, in run
BUILDSTDERR:     asyncio.gather(*tasks, loop=self.loop))
BUILDSTDERR:   File "uvloop/loop.pyx", line 1449, in uvloop.loop.Loop.run_until_complete
BUILDSTDERR:     raise RuntimeError('Event loop stopped before Future completed.')
BUILDSTDERR: RuntimeError: Event loop stopped before Future completed.
BUILDSTDERR: ======================================================================
BUILDSTDERR: FAIL: test_write_pipe (test_pipes.Test_UV_Pipes)
BUILDSTDERR: ----------------------------------------------------------------------
BUILDSTDERR: Traceback (most recent call last):
BUILDSTDERR:   File "/builddir/build/BUILD/uvloop-0.12.2/uvloop/_testbase.py", line 102, in tearDown
BUILDSTDERR:     self.fail('unexpected calls to loop.call_exception_handler()')
BUILDSTDERR: AssertionError: unexpected calls to loop.call_exception_handler()
BUILDSTDERR: ----------------------------------------------------------------------
```

Probably it is related to us using 19.0.0 PyOpenSSL, but we don't have option.

Here is the build log with async debug: https://kojipkgs.fedoraproject.org//work/tasks/6783/34576783/build.log

--- a/tests/test_tcp.py
+++ b/tests/test_tcp.py
@@ -2203,6 +2203,8 @@ class _TestSSL(tb.SSLTestCase):
         sslctx.use_privatekey_file(self.ONLYKEY)
         sslctx.use_certificate_chain_file(self.ONLYCERT)
         client_sslctx = self._create_client_ssl_context()
+        if hasattr(ssl, 'OP_NO_TLSv1_3'):
+            client_sslctx.options |= ssl.OP_NO_TLSv1_3
 
         def server(sock):
             conn = openssl_ssl.Connection(sslctx, sock)
@@ -2560,6 +2562,8 @@ class _TestSSL(tb.SSLTestCase):
         sslctx_openssl.use_privatekey_file(self.ONLYKEY)
         sslctx_openssl.use_certificate_chain_file(self.ONLYCERT)
         client_sslctx = self._create_client_ssl_context()
+        if hasattr(ssl, 'OP_NO_TLSv1_3'):
+            client_sslctx.options |= ssl.OP_NO_TLSv1_3
 
         future = None
 

